Update README.md (#11)
hashicorp/setup-terraform: README.md

        body: output
      })
```

> Specify terraform working directory and previous outputs can be used in subsequent steps for a concise comment on the pull request

Reviewer: There should be an empty line before and after this line. I also think we should just replace the existing commenting example right above it instead of adding a new one. Commenting on pull requests isn't the focus of this GitHub action so I'd like to keep the commenting examples to a minimum in the README.

```suggestion

Outputs can be used in subsequent steps to comment on the pull request.

```

Author: Ok, reverted it back to your suggestion above

Reviewer: Oh sorry! I meant that you can remove the existing example that's right above and replace it with your new example so that there's only one commenting example in the README.

Author: ok, I have replaced the existing example with my new example

```yaml
env:
    tf_actions_working_dir: <Terraform working directory>
steps:
- uses: actions/checkout@v2
- uses: hashicorp/setup-terraform@v1

- name: Terraform fmt
  id: fmt

>   run: cd ${{ env.tf_actions_working_dir }} && terraform fmt

Reviewer: GitHub Actions has a working directory option. Cleaner than the `&&` syntax.

```suggestion
  working-directory: ${{ env.tf_actions_working_dir }}
  run: terraform fmt
```

Author: To make it even more cleaner
```
defaults:
  run:
    working-directory: ${{ env.tf_actions_working_dir }}
steps:
```

  continue-on-error: true

- name: Terraform Init
  id: init

>   run: cd ${{ env.tf_actions_working_dir }} && terraform init

Reviewer: See my previous comment about using the `working-directory` option instead.

- name: Terraform Validate
  id: validate

>   run: cd ${{ env.tf_actions_working_dir }} && terraform validate -no-color

Reviewer: See my previous comment about using the `working-directory` option instead.

- name: Terraform Plan
  id: plan

>   run: cd ${{ env.tf_actions_working_dir }} && terraform plan -no-color

Reviewer: See my previous comment about using the `working-directory` option instead.

  continue-on-error: true

- uses: actions/github-script@0.9.0
  if: github.event_name == 'pull_request'
  env:
    PLAN: "terraform\n${{ steps.plan.outputs.stdout }}"
  with:
    github-token: ${{ secrets.GITHUB_TOKEN }}
    script: |

>       const output = `#### Terraform Format and Style 🖌\`${{ steps.fmt.outcome }}\`

Reviewer: I'm not sure where the `outcome` syntax is coming from. Can you clarify?

Author: when an `id` is given to step, the status of that step is in JSON and it can be accessed by `${{ steps.<Step Id>.outcome }}` E.g
Below is a sample of a step that ran successfully
```
{
  outputs: {},
  outcome: success,
  conclusion: success
}
```
Below is a sample of how a step that failed but `continue-on-error: true` looks like
```
{
  outputs: {},
  outcome: failure,
  conclusion: success
}
```
As you can see from the above, the step failed but it can be accessed with `${{ steps.<Step Id>.outcome }}`. Since the step was set to continue on error, the `conclusion: success` but the step `outcome: failure`

Reviewer: Oh wow. That's pretty hidden in the docs. Found it [here](https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#steps-context).

>       #### Terraform Initialization ⚙️\`${{ steps.init.outcome }}\`

Reviewer: See previous comment regarding `outcome` syntax.

      #### Terraform Validation 🤖${{ steps.validate.outputs.stdout }}

>       #### Terraform Plan 📖\`${{ steps.plan.outcome }}\`

Reviewer: See previous comment regarding `outcome` syntax.
